apps/pwm_test/pkg.yml

```diff
@@ -0,0 +1,35 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+pkg.name: apps/pwm_test
+pkg.type: app
+pkg.description: Basic example application which blinks an LED.
+pkg.author: "Apache Mynewt <dev@mynewt.incubator.apache.org>"
+pkg.homepage: "http://mynewt.apache.org/"
+pkg.keywords:
+
+pkg.deps:
+    - kernel/os
+   # - hw/hal
+   # - hw/cmsis-core
+    - sys/console/full
+   # - hw/bsp/nrf52dk
+    - hw/drivers/pwm
+   # - "@mynewt_nordic/hw/mcu/nordic_sdk"
+    - "@mynewt_nordic/hw/drivers/pwm/pwm_nrf52"
```

Comment:
I would remove the dependency on pwm_nrf52, and move it to hw/bsp/<your_bsp>/pkg.yml instead.
Also I would do the os_dev_create() within that BSPs hal_bsp_init(). Make the creation of the device to be a syscfg within the BSP. That would allow you to use this same app with BSPs which use some other MCU (once PWM drivers for those architectures appear :).